cumulusci/tasks/salesforce/custom_settings.py

```diff
@@ -0,0 +1,124 @@
+            elif isinstance(settings_data, list):
+                for setting_instance in settings_data:
+                    query = None
+
+                    if "location" in setting_instance:
+                        if "profile" in setting_instance["location"]:
+                            # Query for a matching Profile to assign the Setup Owner Id.
+                            profile_name = setting_instance["location"]["profile"]
+                            query = (
+                                f"SELECT Id FROM Profile WHERE Name = '{profile_name}'"
+                            )
+                        elif "user" in setting_instance["location"]:
+                            if "name" in setting_instance["location"]["user"]:
+                                # Query for a matching User to assign the Setup Owner Id.
+                                user_name = setting_instance["location"]["user"]["name"]
+                                query = f"SELECT Id FROM User WHERE Username = '{user_name}'"
+                            elif "email" in setting_instance["location"]["user"]:
+                                # Query for a matching User to assign the Setup Owner Id.
+                                email_address = setting_instance["location"]["user"][
+                                    "email"
+                                ]
+                                query = f"SELECT Id FROM User WHERE Email = '{email_address}'"
+                        elif "org" == setting_instance["location"]:
+                            # Assign the Setup Owner Id to the organization.
+                            query = "SELECT Id FROM Organization"
+
+                    if query is None:
+                        raise CumulusCIException(
+                            f"No valid Setup Owner assignment found for Custom Setting {custom_setting}"
+                        )
+
+                    matches = self.sf.query(query)
+                    if matches["totalSize"] != 1:
+                        raise CumulusCIException(
+                            f"{matches['totalSize']} records matched the settings location query {query}. Exactly one result is required."
+                        )
+
+                    setup_owner_id = matches["records"][0]["Id"]
+
+                    # We can't upsert on SetupOwnerId. Query for any existing records.
+                    existing_records = self.sf.query(
+                        f"SELECT Id FROM {custom_setting} WHERE SetupOwnerId = '{setup_owner_id}'"
+                    )
+
+                    setting_instance["data"].update({"SetupOwnerId": setup_owner_id})
+                    if existing_records["totalSize"] == 0:
+                        proxy_obj.create(setting_instance["data"])
+                    else:
+                        proxy_obj.update(
+                            existing_records["records"][0]["Id"],
+                            setting_instance["data"],
+                        )
```

Comment:
Can you add logging of which settings objects are being updated in which location? I think that'll be useful for debugging.